app/models/team.rb

```diff
@@ -7,5 +7,8 @@ class Team < ApplicationRecord
 
   validates :name, presence: true
 
-  delegate :owner, to: :tournament
+  def owner
+    match_scores.first.owner
+    # this will produce errors if we make teams reusable
```

Comment:
... and for matches without match scores. is there another way to implement this?
imo the owner attribute should only be delegated up, not down, because that will just introduce unwanted issues/dependencies

Reply:
There should never be a match without match_scores that has teams. How would that even work. A matches teams are defined by the two match_scores hanging on it.

Reply:
At the moment this is entirely possible, because it is not enforced/validated in any way. We shouldn't rely on the existence of that attribute (and it's still delegating the attribute down).

Imo we should reconsider this once we have teams not belonging to a tournament and leave it as is for now.